### ISSUE TITLE
Migrate User Guide and Developer Guide to MarkBind site

### ISSUE THREAD

**Issue by nicholaschuayunzhi** [MarkBind/markbind#180]
We should migrate our documentation to our MarkBind site: https://markbind.github.io/markbind/ because documentation can be committed along with each feature.

This ensures that documentation is updated regularly and its to review changes to the documentation (instead of pushing directly to the wiki).

Things to note:
We need a way to preview the newly built site.
I've found a way we can use Netlify in https://github.com/nicholaschuayunzhi/markbind/pull/1 
![image](https://user-images.githubusercontent.com/22221132/37871946-07ac0b12-302e-11e8-8aff-bdce3aeff137.png) but it needs the baseUrl of the site to be `baseUrl: ""`. Not too sure how we can manage this.

**Comment by damithc** [comment]
This can be useful. BTW, netlify preview of nicholaschuayunzhi#1 shows 'Not Found'?

**Comment by nicholaschuayunzhi** [comment]
> BTW, netlify preview of nicholaschuayunzhi#1 shows 'Not Found'?

Ahh my bad. I was playing around with the site name which broke the deployment. I have deployed it again and it should be viewable on https://deploy-preview-1--markbind.netlify.com/

**Comment by damithc** [comment]
One approach:
* Keep dev guide in markdown format i.e., updates as soon as a PR is merged
* Keep user guide in markbind format and deploy to gh-pages only when a new version is released. 

I wonder if there is a way to keep older versions of the user guide also online, in case a user doesn't want to update immediately.

Any other approaches?

**Comment by acjh** [comment]
> I wonder if there is a way to keep older versions of the user guide also online, in case a user doesn't want to update immediately.

We can use different branches and RawGit: https://rawgit.com/MarkBind/markbind/gh-pages/

Changing the `baseUrl` seems unavoidable; it's tracked in #63 (Support staging multiple PRs).
As you mentioned there, perhaps the `baseUrl` can be provided **as a command line parameter**.
RawGit is branch-specific, but it seems possible to always pass an empty string to Netlify for PRs.
As for version-specific branches on RawGit, it's reasonable to tweak it manually when deploying.

Anyway, there are few backward-incompatible changes, so version-specific branches should wait.

**Comment by damithc** [comment]
> Anyway, there are few backward-incompatible changes, so version-specific branches should wait.

Yes, no hurry. At the moment we can require users to always use the most recent version.

**Comment by damithc** [comment]
Can set up netlify for this repo now?
